lib/Sema/CSSolver.cpp

```diff
@@ -2550,15 +2514,18 @@ bool ConstraintSystem::solveSimplified(
     //        already have a solution involving non-generic operators,
     //        but continue looking for a better non-generic operator
     //        solution.
-    if (firstNonGenericOperatorSolution &&
-        isGenericOperatorOrUnavailable(constraint))
-      continue;
+    if (firstNonGenericOperatorSolution) {
+      if (currentChoice.isGenericOperatorOrUnavailable())
+        continue;
+    }
```

Comment:
Ah, I will change it back to one if, I just yenked couple of things at the last moment to the separate patch.